src/Data/Optional.hs

```diff
@@ -147,3 +149,18 @@ instance Fractional a => Fractional (Optional a) where
     recip = fmap recip
 
     (/) = liftA2 (/)
+
+-- | The 'optional' function takes a default value, a function, and an
+-- 'Optional' value. If the 'Optional' value is 'Default', the function returns
+-- the default value. Otherwise, it applies the function to the value inside the
+-- 'Optional' and returns the result.
+optional :: b -> (a -> b) -> Optional a -> b
+optional n _ Default  = n
+optional _ f (Specific x) = f x
+
+-- | The 'fromOptional' function takes a default value and an 'Optional'
+-- value.  If the 'Optional' is 'Default', it returns the default value;
+-- otherwise, it returns the value contained in the 'Optional'.
+fromOptional :: a -> Optional a -> a
```

Comment:
How about naming this `defaultTo` instead?

I would like to reserve `fromOptional` for this function:

```haskell
fromOptional :: Alternative f => Optional a -> f a
fromOptional  Default     = empty
fromOptional (Specific x) = pure x